merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/LocalMissionModelService.java

```diff
@@ -141,20 +141,26 @@ public List<ValidationNotice> validateActivityArguments(final String missionMode
    * @return A map of validation errors mapping activity instance ID to failure message. If validation succeeds the map is empty.
    */
   @Override
-  public Map<ActivityInstanceId, ActivityInstantiationFailure> validateActivityInstantiations(final String missionModelId, final Map<ActivityInstanceId, SerializedActivity> activities)
+  public Map<ActivityInstanceId, ActivityInstantiationFailure>
+  validateActivityInstantiations(final String missionModelId,
+                                 final Map<ActivityInstanceId, SerializedActivity> activities)
   throws NoSuchMissionModelException, MissionModelLoadException
   {
+    final var factory = this.loadMissionModelFactory(missionModelId);
+    final var registry = DirectiveTypeRegistry.extract(factory);
+
     final var failures = new HashMap<ActivityInstanceId, ActivityInstantiationFailure>();
 
     for (final var entry : activities.entrySet()) {
       final var id = entry.getKey();
       final var act = entry.getValue();
       try {
-        this.getActivityEffectiveArguments(missionModelId, act); // The return value is intentionally ignored - we are only interested in failures
+        // The return value is intentionally ignored - we are only interested in failures
+        final var specType = Optional
+        .ofNullable(registry.taskSpecTypes().get(act.getTypeName()))
+        .orElseThrow(() -> new MissionModelService.NoSuchActivityTypeException(act.getTypeName()));
```

Comment:
After inlining `getActivityEffectiveArguments`, we seem to have dropped the call to `specType.getEffectiveArguments(activity.getArguments());`, I think we still want that, since without it we're not examining the arguments of the activity at all as part of this validation.

Reply:
Very good point. Thank you!